src/front-end/graphics.js

```diff
@@ -487,187 +530,199 @@ var Render = (function(){
-             var radiusar = properties.radiusar
-             var arcPath = describeArc(dx + xar, dy + yar, radiusar, rotationar,
-                 anglear < 0 ? (rotationar - anglear) : (rotationar + anglear));
-                 var arc = s.path(arcPath);
-                 arc.attr({
-                     "fill-opacity": 0,
-                     stroke: Utils.hex(color[0], color[1], color[2]),
-                     strokeWidth: 2
-                 });
-                 if(style == "line"){
-                 } else if (style == "wedge") {
-                     var pf = describeFullArc(dx + xar, dy + yar, radiusar, rotationar,
-                         anglear < 0 ? (rotationar - anglear) : (rotationar + anglear));
-                         var f = s.path(pf);
-                         f.attr({
-                             fill: Utils.hex(0, 0, 205),
-                             "fill-opacity": 0.6,
-                         });
-                         var g = s.g(arc,f)
-                     }
-                 }
-     }
-
-    /**
-     * Renders a parallelogram on the canvas
-     * @param       {Snap} s  Snap.svg global object
-     * @param       {JSON} obj JSON object from Haskell server
-     * FIXME: refactor
-     * FIXME: NOT PORTED
-     */
-     function _renderParallelogram(s, obj) {
-         // TODO fix this!
-         [sx, sy] = Utils.scr([properties.xpa, properties.ypa])
-         var sizeX = properties.sizeXpa;
-         var sizeY = properties.sizeYpa;
-         var angle = properties.anglepa;
-         var rotation = properties.rotationpa
-         var path = ""
-         if(angle <= 90.0){
-             path = "M " + (-(sizeX/2) - sizeY/getTanFromDegrees(angle)) + " " + (-(sizeY/2)) +  " L " +
-             (-(sizeX/2)) + " " + (sizeY/2) +  " L " +  (sizeX/2) + " " + (sizeY/2) +  " L " +
-             ((sizeX/2) - (sizeY/getTanFromDegrees(angle))) + " " + (-(sizeY/2))
-         }
-         else{
-             path = "M " + (-(sizeX/2)) + " " + (sizeY/2) +  " L " +
-             (sizeX/2) + " " + (sizeY/2) +  " L " + ((sizeX/2) + sizeY/getTanFromDegrees(angle)) + " " + (-(sizeY/2))
-             +  " L " + (-(sizeX/2) + (sizeY/getTanFromDegrees(angle))) + " " + (-(sizeY/2))
-         }
-
-         var myMatrix = new Snap.Matrix();
-         myMatrix.translate(sx, sy);
-         myMatrix.rotate(rotation, 0, 0);
-         var parallelogram = s.path(path).transform(myMatrix.toTransformString())
-         parallelogram.data("name", properties.namepa)
-         var color = properties.colorpa;
-         parallelogram.attr({
+        var myMatrix = new Snap.Matrix();
+        myMatrix.translate(0, 0);
+        myMatrix.rotate(rotation, x, y);
+        if (style == "wedge") {
+          var f = s.rect(x,y - size ,size,size).transform(myMatrix.toTransformString());
+          f.attr({
+            fill: Utils.hex(0, 0, 205),
+            "fill-opacity": 0.6,
+          });
+          return f;
+        } else{
```

Comment:
is the responsibility on the Style writer to decide when a mark is perpendicular or not?

Reply:
this comment should really be about line 626

Reply:
By specifying the `isRight` property, I thinks this is pretty clear but I'll add it to the shapes doc